trio/_core/_run.py

```diff
@@ -1540,7 +1543,7 @@ def task_exited(self, task, outcome):
     ################
 
     @_public
-    def spawn_system_task(self, async_fn, *args, name=None):
+    def spawn_system_task(self, async_fn, *args, name=None, context=None):
```

Comment:
I don't think this is supposed to be here anymore? System tasks should always inherit the system context, right?

Reply:
I actually haven't been able to make tests pass without it, I've been trying several different approaches, copying the context in different places, but in some I get errors that show it's not copying the right context, or thread timeout errors, or it breaks internals. :scream: I suspect it could be just that I lack the knowledge of the internals to do it in a different and better way.

Do you see a way that could be solved to remove it?

On the other hand, I understand @richardsheridan thinks this could be useful... for me it's the same, but I just don't see how to make it work without this. :thinking: